testing/chain.go

```diff
@@ -324,6 +355,14 @@ func (chain *TestChain) SendMsgs(msgs ...sdk.Msg) (*sdk.Result, error) {
 	if err != nil {
 		return nil, err
 	}
+	// store packets sent during the execution of this transaction
+	for _, event := range r.Events {
+		if event.Type == channeltypes.EventTypeSendPacket {
+			packet, err := channelkeeper.ReconstructPacketFromEvent(event)
+			require.NoError(chain.T, err)
+			chain.SentPackets[packet.Sequence] = packet
+		}
```

Comment:
The same code is three times, it might be beneficial to make a reusable function instead.